R/particle_filter.R

```diff
@@ -188,13 +256,16 @@ particle_filter <- R6::R6Class(
         stop("Expected first 't' element to be zero")
       }
       step <- private$data[[private$n_steps]]$step_end + t
-      n_particles <- ncol(self$state)
-      res <- private$last_model$run(step, self$state, replicate = n_particles,
-                                    use_names = FALSE, return_minimal = TRUE)
-      res <- aperm(res, c(1, 3, 2))
+
+      res <- array(NA_real_, dim = c(dim(self$state), length(step) - 1))
+      for (t in step[-1]) {
+        private$last_model$run(t)
+        res[, , t - step[1]] <- private$last_model$state()
```

Comment:
is this assignment (`t - step[1]`) correct for steps with gaps in it (e.g. step as 0, 4, 8, ...)?